### PR TITLE
Updates Kube RBAC for Watching Envoy Deployment

### LINKED ISSUE #397: Controller Fails to List Deployments
https://github.com/envoyproxy/gateway/pull/392 added support for watching managing status based on the availability of the Envoy deployment. However, RBAC was not updated to allow the controller to list Deployments:

```
W0920 17:43:07.490750       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:envoy-gateway-system:envoy-gateway" cannot list resource "deployments" in API group "apps" at the cluster scope
```

### PR DESCRIPTION
**Pull request by danehans**
Fixes #397 

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

File: internal/provider/kubernetes/rbac.go [hunk 1]
```diff
@@ -5,3 +5,4 @@ package kubernetes
 
 // RBAC for watched resources of Gateway API controllers.
 // +kubebuilder:rbac:groups="",resources=secrets;services;namespaces,verbs=get;list;watch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch
```
Comment (arkodg) [inline internal/provider/kubernetes/rbac.go:8]: shouldnt this be limited to the `envoy-gateway-system` namespace ?

Comment (danehans) [reply internal/provider/kubernetes/rbac.go:8]: As with the other [watched resources](https://github.com/envoyproxy/gateway/blob/main/internal/provider/kubernetes/rbac.go#L6-L7), it's my understanding a controller [watch](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller#Controller.Watch) is across all namespaces and then an eventhandler/predicate are used to filter the sources, e.g. Deployment.

EG has RBAC config split between the Infra Manager and the Kube provider. In this case, the Kube provider controller is the component that requires a Deployment `watch` verb and not the Infra Manager.

@skriss thoughts?

Comment (danehans) [reply internal/provider/kubernetes/rbac.go:8]: When I add the `watch` and `list` verbes to the Infra Manager RBAC, I still see this bug:
```
$ kubectl get role/infra-manager -n envoy-gateway-system -o yaml
...
  resources:
  - deployments
  verbs:
  - create
  - get
  - list
  - watch
  - update
  - delete
```

EG Logs:
```
E0920 18:48:20.575121       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1.Deployment: failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:envoy-gateway-system:envoy-gateway" cannot list resource "deployments" in API group "apps" at the cluster scope
```

@arkodg `... at the cluster scope` is key to the error message.

Comment (skriss) [reply internal/provider/kubernetes/rbac.go:8]: @danehans that sounds right to me, I don't think it's possible to constraint the informer for a single resource type to just one namespace.

**Review by skriss — APPROVED** [review]

File: internal/provider/kubernetes/rbac.go (hunk 1 above)
Comment (arkodg) [reply internal/provider/kubernetes/rbac.go:8]: rbac can be kept within infra-manager when https://github.com/envoyproxy/gateway/issues/371 is worked on

**Pull request merged**